### PR TITLE
Adds support and visibility for Databend connection

### LINKED ISSUE #35640: Support Databend database
Here is JDBC driver
https://github.com/datafuselabs/databend-jdbc
And here is previous attempt PR https://github.com/dbeaver/dbeaver/pull/19558

### PR DESCRIPTION
**Pull request by rad-pat**
Further to previous PR https://github.com/dbeaver/dbeaver/pull/19558, I made some fixes and this seems to be working for me.

### CODE REVIEW

**Comment by E1izabeth** [comment]
Thank you for contribution. We'll review your PR and let you know if some changes are required

File: plugins/org.jkiss.dbeaver.ext.databend/src/org/jkiss/dbeaver/ext/databend/model/DatabendConstants.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package org.jkiss.dbeaver.ext.databend.model;
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.databend/src/org/jkiss/dbeaver/ext/databend/model/DatabendConstants.java:17]: Please add copyright

File: plugins/org.jkiss.dbeaver.ext.databend/src/org/jkiss/dbeaver/ext/databend/model/DatabendSSLImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.databend.model;
+
+import org.jkiss.dbeaver.model.impl.net.SSLHandlerImpl;
+
+public class DatabendSSLImpl extends SSLHandlerImpl {
+
+}
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.databend/src/org/jkiss/dbeaver/ext/databend/model/DatabendSSLImpl.java]: Why do you need this class?

File: plugins/org.jkiss.dbeaver.ext.databend/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+Bundle-Name: DBeaver Databend Model
+Bundle-SymbolicName: org.jkiss.dbeaver.ext.databend;singleton:=true
+Bundle-Version: 1.0.0.qualifier
+Bundle-Release-Date: 20240917
+Require-Bundle: org.eclipse.core.runtime,
+ org.jkiss.dbeaver.model,
+ org.jkiss.dbeaver.ext.generic;visibility:=reexport,
+ com.google.guava;resolution:=optional,
+ com.google.gson,
+ org.jkiss.bundle.antlr4
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.databend/META-INF/MANIFEST.MF]: I believe you don't need these bundles dependencies

File: plugins/org.jkiss.dbeaver.ext.databend.ui/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+Bundle-SymbolicName: org.jkiss.dbeaver.ext.databend.ui;singleton:=true
+Bundle-Version: 1.0.0.qualifier
+Bundle-Release-Date: 20240917
+Require-Bundle: org.eclipse.ui,
+ org.eclipse.core.runtime,
+ org.jkiss.dbeaver.model,
+ org.jkiss.dbeaver.ext.databend,
+ org.jkiss.dbeaver.ui,
+ org.jkiss.dbeaver.ui.editors.connection,
+ org.jkiss.dbeaver.ui.editors.data,
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ext.databend.ui/META-INF/MANIFEST.MF]: Do you really need these dependencies?

**Comment by E1izabeth** [comment]
Also LICENSE file is empty

**Comment by rad-pat** [comment]
Thanks @E1izabeth , I'll attend to these as soon as I can.

**Comment by rad-pat** [comment]
Apologies, I've been unable to progress with this as I can no longer build & run to test within Eclipse after rebasing to make changes. I asked in a discussion https://github.com/orgs/dbeaver/discussions/35703, but no-one has replied.

**Comment by E1izabeth** [comment]
@rad-pat maybe you could try using IDEA for development? Here is the guide https://github.com/dbeaver/dbeaver/wiki/Develop-in-IDEA

**Review by serge-rider — APPROVED** [review]

**Comment by rad-pat** [comment]
> @rad-pat maybe you could try using IDEA for development? Here is the guide https://github.com/dbeaver/dbeaver/wiki/Develop-in-IDEA

Thanks, I tried IDEA but I couldn't get it to work either. Then I tried going back to Eclipse and it was working again. Absolutely no idea why, but at least I could debug and test it. Thanks for the help

**Comment by E1izabeth** [comment]
Thanks

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
1. No icons for database are shown 
image
image

2. No statistic tab shown in SQL editor after e.g. CREATE TABLE, INSERT INTO script execution. Only No Data tab is shown.
image

3. Error appears on attempt to open function properties tab
image
```

eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
Command-line arguments:  -os win32 -ws win32 -arch x86_64

This is a continuation of log file C:\Users\Dziyana\AppData\Roaming\DBeaverData\workspace9\.metadata\.bak_0.log
Created Time: 2024-11-19 10:42:47.118

org.jkiss.dbeaver.model
Error
Tue Nov 19 10:45:31 CET 2024
Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null

java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null
	at java.base/java.util.regex.Matcher.getTextLength(Unknown Source)
	at java.base/java.util.regex.Matcher.reset(Unknown Source)
	at java.base/java.util.regex.Matcher.<init>(Unknown Source)
	at java.base/java.util.regex.Pattern.matcher(Unknown Source)
	at org.jkiss.dbeaver.ext.generic.model.GenericProcedure.loadProcedureColumns(GenericProcedure.java:227)
	at org.jkiss.dbeaver.ext.generic.model.GenericProcedure.getParameters(GenericProcedure.java:124)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.extractPropertyValue(DBNDatabaseNode.java:1047)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode$PropertyValueReader.run(DBNDatabaseNode.java:1119)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode$PropertyValueReader.run(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadTreeItems(DBNDatabaseNode.java:664)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadChildren(DBNDatabaseNode.java:475)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:229)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.model.navigator.DBNUtils.getNodeChildrenFiltered(DBNUtils.java:94)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ItemListControl$ItemLoadService.evaluate(ItemListControl.java:306)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ItemListControl$ItemLoadService.evaluate(ItemListControl.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:88)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:119)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by rad-pat** [comment]
Thanks for this @uslss, I will look into them ASAP. Missing icon is weird, I'm pretty sure I should've picked that up 🤷

**Comment by rad-pat** [comment]
On Item 2, Statistics not shown, that appears to be because JDBC driver returns -1 for getUpdateCount(). Perhaps the JDBC driver is incorrect? It currently returns an empty result set, perhaps it should not? With the number of rows updated being -1, it fails this test and thus does not show the statistics panel.
https://github.com/dbeaver/dbeaver/blob/70be065bae00fff8710298cefcec46027972fe38/plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/execute/SQLQueryJob.java#L799

**Comment by rad-pat** [comment]
@uslss , @E1izabeth , are you happy with the fixes made?

**Comment by E1izabeth** [comment]
Thank you for the corrections. We'll review them shortly

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Comment by arhayka** [comment]
Hello @rad-pat!
Thank you for being with DBeaver! We appreciate your support and want to send you a compliment. Please write to an email [anna.pozdeeva@dbeaver.com](mailto:anna.pozdeeva@dbeaver.com) and provide your nickname in GitHub and shipping details: Name and Surname, Email, Contact phone, Full Address (apt, street, city, state, country, post code).

**Pull request merged**